frontend/src/canvas/CanvasContainer.js

```diff
@@ -259,6 +259,17 @@ const CanvasContainer = (props) => {
     if (props.selectedPositionX === null || props.selectedPositionY === null) {
       return null;
     }
+
+    const existingPixel = props.extraPixelsData.find(
+      (pixel) =>
+        pixel.x == props.selectedPositionX && pixel.y == props.selectedPositionY
+    );
+
+    if (existingPixel) {
+      props.setSelectedColorId(existingPixel.colorId);
```

Comment:
We don't want to change the selected color id here, but use the color in the extra pixels data to return the inverse like this 
```
return (
        '#' +
        (255 - color[0]).toString(16).padStart(2, '0') +
        (255 - color[1]).toString(16).padStart(2, '0') +
        (255 - color[2]).toString(16).padStart(2, '0') +
        color[3].toString(16).padStart(2, '0')
      );```